src/halmos/sevm.py

```diff
@@ -1348,7 +1356,35 @@ def arith(self, ex: Exec, op: int, w1: Word, w2: Word) -> Word:
             return w1 - w2
 
         if op == EVM.MUL:
-            return w1 * w2
+            if is_bv_value(w1) and is_bv_value(w2):
+                return w1 * w2
```

Comment:
just to double check: we're fine here in case of overflow?

Reply:
yes, and it's indeed intended.